codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -74,6 +77,9 @@ public HttpHeaders add(HttpHeaders headers) {
     @Override
     public HttpHeaders set(HttpHeaders headers) {
         if (headers instanceof DefaultHttpHeaders) {
+            if (headers == this) {
+                throw new IllegalArgumentException("can't add to itself.");
```

Comment:
@normanmaurer @nmittler - Well now things get interesting :)

I'm not strong on this either and it seems like no one is so @normanmaurer why don't you decide and just make sure we handle it consistently across versions (and documented).

Just to clarify the programming error statement ... the following code (and any code that calls `set` where it may take the same object) has a programming error:
```java
HttpHeaders a = ...;
doSomething(a, someMethodWhichMayReturnParamter(a));

void doSomething(HttpHeaders a, HttpHeaders b) {
  if (something happens) {
    // Programming error because no check to a != b?
    a.set(b);
  }
}
```